editor/effects.js

```diff
@@ -268,12 +268,22 @@ export default {
 		dispatch( savePost() );
 	},
 	SETUP_EDITOR( action ) {
-		const { post } = action;
+		const { post, settings } = action;
 		const effects = [];
 
 		// Parse content as blocks
 		if ( post.content.raw ) {
 			effects.push( resetBlocks( parse( post.content.raw ) ) );
+		} else if ( settings.template ) {
+			const blocks = map( settings.template.blocks, ( { name, attributes } ) => {
```

Comment:
Are we sure we are going to want additional properties on the `template` object instead of making `template = template.blocks`?

Reply:
I don't know honestly. Server-side I think so, because templates could have a name/description but client-side I don't really know. I thought we should do this to avoid future breaking changes. Templates could have a name

Reply:
Yes, just thinking out-loud. There are things like "locking" which might be interesting there. Not sure about naming templates, but maybe an `id / slug` if they are stored somewhere?

Reply:
Thinking further, I can't see why we would pair defining blocks and naming in the CPT. If anything, the "template" would be defined outside the CPT and referenced here by slug.

Also, things like locking could be defined as another attribute of the CPT registration:

```php
'template' => array(
	array( 'core/image' ),
	array( 'core/paragraph', array(
		'placeholder' => 'Add a book description',
	) ),
	array( 'core/quote' ),
),
'template_lock' => true,
```

Finally, what do you think of simplifying the syntax to not require `name => 'core/heading'? I think we can rely on the definition of a block to be `array( blockName, attributes ... )`.

Reply:
Sure, I was not sure about the extra keys anyway.